Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Infrastructure/DrugClient.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Text.Json;
+
+namespace HealthcareHospitalManagementSystem.Services
+{
+    public class DrugClient
+    {
+        private readonly HttpClient _httpClient;
+
+        public DrugClient(HttpClient httpClient)
+        {
+            _httpClient = httpClient;
+        }
+
+        public async Task<DrugReportClass> GetDrugReportAsClassAsync(string searchTerm, CancellationToken cancellationToken = default)
```

Comment:
You do not need to have default value here 
CancellationToken cancellationToken = default

Just CancellationToken cancellationToken, default will be needed if there is no way to take cancellationToken, but in ASP.NET framework it automatically will be place during the call to Controller method.